doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1746,33 +1784,77 @@ Controlling the proof flow
    impossible. In most cases, :g:`P` or :g:`∼P` is one of the hypotheses of the
    local context.
 
-.. tacn:: contradiction
-   :name: contradiction
+.. tacn:: contradiction {? @one_term_with_bindings }
 
-   This tactic applies to any goal. The contradiction tactic attempts to
-   find in the current context (after all intros) a hypothesis that is
-   equivalent to an empty inductive type (e.g. :g:`False`), to the negation of
-   a singleton inductive type (e.g. :g:`True` or :g:`x=x`), or two contradictory
-   hypotheses :g:`P` and :g:`not P`.
+   Tries to prove the current goal by finding a contradiction.
 
-   .. exn:: No such assumption.
-      :undocumented:
+   If :n:`@one_term_with_bindings` is not provided, the tactic first does an
+   :tacn:`intros`.  The tactic then proves the goal if the updated context has
+   a pair of hypotheses where one is the negation of the other (e.g. :n:`P` and
+   not :n:`~P`) or if there is a hypothesis with the type :n:`False` (or another
+   empty inductive type).
```

Comment:
It looks like we lost the third option "the negation of a singleton inductive type (e.g. True or x=x)" from the previous doc in this paragraph.

Reply:
OK.  However `1=0` appears to be a singleton inductive and `~ (1 = 0)` is a true and provable statement.  Can you explain?

Reply:
That's a good point. The previous description was imprecise, but I don't really know how to fix it. Jason's proposal in https://github.com/coq/coq/pull/16659#discussion_r1039512423 was more correct than the previous doc. From your answer to this comment, you seem to think you have some better way of expressing this, but I cannot check it because you forgot to push your changes.

Reply:
Regarding `(1=0)`, see [here](https://github.com/coq/coq/pull/16659#discussion_r1055977206) and respond there.